sqlalchemy_redshift/commands.py

```diff
@@ -18,10 +18,21 @@
 ACCESS_KEY_ID_RE = re.compile('[A-Z0-9]{20}')
 SECRET_ACCESS_KEY_RE = re.compile('[A-Za-z0-9/+=]{40}')
 TOKEN_RE = re.compile('[A-Za-z0-9/+=]+')
+IAM_ROLE_RE = re.compile('arn:aws:iam::[0-9]+:role/.+')
 
 
-def _process_aws_credentials(access_key_id, secret_access_key,
-                             session_token=None):
+def _process_aws_credentials(access_key_id=None, secret_access_key=None,
+                             session_token=None, iam_role=None):
+
+    if iam_role is not None:
+        if not IAM_ROLE_RE.match(iam_role):
+            raise ValueError(
+                'invalid IAM role ARN; does not match {pattern}'.format(
+                    pattern=IAM_ROLE_RE.pattern,
+                )
+            )
+        else:
+            return 'aws_iam_role=' + iam_role
```

Comment:
The way this is written, `iam_role` takes precedence over `access_key_id`/`secret_access_key`/`session_token`, but the documentation doesn't make this explicit. Providing both types of credentials seems like it indicates confusion on the part of the user, and I'd propose that we raise an error in that case indicating that the two credential styles are mutually exclusive and should not be mixed.

So, if within this `if` block, we'd check if any of access key ID, secret access key, or session token are not `None`, and then raise an error.